File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        
        int n=nums.size();
        vector<int> prefix(n);
        vector<int> suffix(n);
        vector<int> res(n);
        

        for(int i=0;i<nums.size();i++){

           if(i!=0)
              prefix[i]=nums[i-1]*prefix[i-1];
           else 
              prefix[i]=1;              
        
        }

        for(int i=nums.size()-1;i>-1;i--){

           if(i!=nums.size()-1)
              suffix[i]=nums[i+1]*suffix[i+1];
           else 
              suffix[i]=1;
        }

        for(int i=0;i<nums.size();i++){

         res[i]=suffix[i]*prefix[i];             
        
        }

          
           return res;



        }

    
};

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {

        int l=0, r=nums.length-1;

        int mid = (r-l)/2;

        int res=-1;
       

        

        while(l<=r){
 
          if(nums[mid]==target){
            return mid;
          }

          else if(nums[mid]>target){
              r=mid-1;
              mid=l+(r-l)/2;

          }
           if(nums[mid]<target){
               l=mid+1;
               mid=l+(r-l)/2;
          }
        
        
        }

        return res;
        
    }
}
